python/mxnet/io/io.py

```diff
@@ -966,6 +985,165 @@ def creator(*args, **kwargs):
+                    continue
+                yield item
+    data_iter = iter(get_generator())
+    data_batch = list(itertools.islice(data_iter, batch_size))
+    while data_batch:
+        yield data_batch
+        data_batch = list(itertools.islice(data_iter, batch_size))
+
+def _read_worker(q_out, transforms, color, quality, encoding, data_record):
+    """
+    Helper function that will be run by the read workers
+    to fetch the image from the input queue apply
+    transformations and put it into output priority queue.
+    Parameters
+    ----------
+    args: object
+    q_out: queue
+    color: color
+    quality: quality
+    encoding: encoding
+    deq: image instance to work on.
+    """
+    i, item = data_record
+    fullpath = os.path.join(args.root, item[1])
+    try:
+        # construct the header of the record
+        if len(item) > 3:
+            header = recordio.IRHeader(0, item[2:], item[0], 0)
+        else:
+            header = recordio.IRHeader(0, item[2], item[0], 0)
+
+        img = cv2.imread(fullpath, args.color)
+        if img is None:
+            logging.info('imread read blank (None) image for file: %s' % fullpath)
+            return
+        img = transforms(img)
+        s = recordio.pack_img(header, img, quality=args.quality, img_fmt=args.encoding)
+        q_out.put((i, s, item))
+    except Exception as e:
+        logging.info('pack_img error on file: %s' % fullpath, e)
+        return
+
+def _validate_filenames(list_file, output_path):
+    """
+    Helper function to validate the file paths of
+    the input list file and output .rec file path.
+    Parameters
+    --------
+    list_file: input list file path
+    output_path: path to the output directory
+    """
+    if not os.path.isfile(list_file):
+        raise Exception("Input list file is invalid - \
+            1. Wrong filename or file path \n2. List file should be of format *.lst")
+    if not os.path.isdir(output_path):
+        raise Exception("Output path should be a directory where the \
+            rec files will be stored.")
+
+def _count_elem(iter):
+    """
+    Helper function to count the number of elements in
+    a generator.
+    Parameters
+    -----
+    iter: generator object
+    Returns
+    -----
+    count: total count of elements
+    """    
+    cnt = itertools.count()
+    deque(zip(iter, cnt), 0)
+    return next(cnt)
+
+def im2rec(list_file, transforms, dataset_params, output_path):
+    """
+    API to convert the input image dataset into recordIO file format
+    """
+    _validate_filenames(list_file, output_path)
+    fname = os.path.basename(list_file).split('.')[0]
+    shared_obj_mgr = SharedObjectManager()
+    shared_obj_mgr.start()
+
+    num_workers = dataset_params.get('num_workers', 63)
+    batch_size = dataset_params.get('batch_size', 4096)
+    parts = dataset_params.get('parts', 1)
+    color = dataset_params.get('color', 1)
+    encoding = dataset_params.get('encoding', '.jpg')
```

Comment:
rename ecoding->output_encoding

Reply:
only output requires encoding, so I think 'encoding' is fine. 
BTW, is there an option to bypass encoding, the existing python code has this functionality